src/OrchardCore.Modules/OrchardCore.Contents/Liquid/FullTextAspectFilter.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Fluid;
+using Fluid.Values;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Models;
+using OrchardCore.Liquid;
+
+namespace OrchardCore.Contents.Liquid
+{
+    public class FullTextAspectFilter : ILiquidFilter
+    {
+        public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, TemplateContext ctx)
+        {
+            if (!ctx.AmbientValues.TryGetValue("Services", out var services))
+            {
+                throw new ArgumentException("Services missing while invoking 'full_text_aspect'");
+            }
+
+            var contentManager = ((IServiceProvider)services).GetRequiredService<IContentManager>();
+
+            if (input.Type == FluidValues.Array)
+            {
+                var contentItems = new List<ContentItem>();
+                foreach(var objValue in input.Enumerate())
+                {
+                    var contentItem = GetContentItem(objValue);
+                    if (contentItem != null)
+                    {
+                        contentItems.Add(contentItem);
+                    }
+                }
+
+                if (!contentItems.Any())
+                {
+                    return NilValue.Instance;
+                }
+
+                var aspects = new List<FullTextAspect>();
+
+                foreach(var contentItem in contentItems)
+                {
+                    aspects.Add(await contentManager.PopulateAspectAsync<FullTextAspect>(contentItem));
+                }
+                
+                // When returning segments seperate them so multiple segments are indexed individually.
+                return new ArrayValue(aspects.SelectMany(x => x.Segments).Where(x => !String.IsNullOrEmpty(x)).Select(x => new StringValue(x + ' ')));
```

Comment:
Seemed the best way to make sure that multiple widgets had their values seperated.
Might be better ways?